cmake/sysbuild/sign_nrf54h20.cmake

```diff
@@ -0,0 +1,265 @@
+
+  string(REGEX MATCH ".*_secondary_app$" is_secondary_app ${main_image})
+
+  # Fetch devicetree details for flash and slot information.
+  dt_chosen(flash_node TARGET mcuboot PROPERTY "zephyr,flash")
+  dt_prop(write_block_size TARGET mcuboot PATH "${flash_node}" PROPERTY "write-block-size")
+  dt_nodelabel(slot0_path TARGET mcuboot NODELABEL "slot0_partition" REQUIRED)
+  dt_reg_addr(slot0_addr TARGET mcuboot PATH ${slot0_path})
+  dt_reg_size(slot0_size TARGET mcuboot PATH ${slot0_path})
+  dt_nodelabel(slot1_path TARGET mcuboot NODELABEL "slot1_partition" REQUIRED)
+  dt_reg_addr(slot1_addr TARGET mcuboot PATH ${slot1_path})
+  dt_reg_size(slot1_size TARGET mcuboot PATH ${slot1_path})
+  if(NOT write_block_size)
+    set(write_block_size 4)
+    message(WARNING "slot0_partition write block size devicetree parameter is missing, assuming write block size is 4")
+  endif()
+
+  # Fetch devicetree details for executable RAM region.
+  dt_chosen(ram_load_dev TARGET mcuboot PROPERTY "mcuboot,ram-load-dev")
+  if(DEFINED ram_load_dev)
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${ram_load_dev})
+  else()
+    dt_chosen(chosen_ram TARGET mcuboot PROPERTY "zephyr,sram")
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${chosen_ram})
+  endif()
+
+  # Fetch devicetree details for the active code partition.
+  dt_chosen(code_flash TARGET ${main_image} PROPERTY "zephyr,code-partition")
+  dt_reg_addr(code_addr TARGET ${main_image} PATH ${code_flash})
+  set(start_offset)
+  sysbuild_get(start_offset IMAGE ${main_image} VAR CONFIG_ROM_START_OFFSET KCONFIG)
+
+  # Append key file path.
+  if(NOT "${keyfile}" STREQUAL "")
+    set(imgtool_args --key "${keyfile}" ${imgtool_args})
+  endif()
+
+  # Construct imgtool command, based on the selected MCUboot mode.
+  set(imgtool_rom_command)
+  if(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT OR SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP)
+    if(is_secondary_app)
+      set(slot_size ${slot1_size})
+    else()
+      set(slot_size ${slot0_size})
+    endif()
+    set(imgtool_rom_command --rom-fixed ${code_addr})
+  else()
+    message(FATAL_ERROR "Only Direct XIP MCUboot modes are supported.")
+    return()
+  endif()
+
+  # Basic 'imgtool sign' command with known image information.
+  set(imgtool_sign ${PYTHON_EXECUTABLE} ${IMGTOOL} sign
+      --version ${CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION} --header-size ${start_offset}
+      --slot-size ${slot_size} ${imgtool_rom_command})
+  set(imgtool_args --align ${write_block_size} ${imgtool_args})
+
+  # Extensionless prefix of any output file.
+  sysbuild_get(BINARY_DIR IMAGE ${main_image} VAR APPLICATION_BINARY_DIR CACHE)
+  sysbuild_get(BINARY_BIN_FILE IMAGE ${main_image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+  if(is_secondary_app)
+    set(output "${BINARY_DIR}/zephyr/${BINARY_BIN_FILE}_secondary_app")
+  else()
+    set(output "${BINARY_DIR}/zephyr/${BINARY_BIN_FILE}")
+  endif()
```

Comment:
output files from sysbuild are placed in the sysbuild folder in ncs, not put into the image folders, and this puts a secondary app into the main app image which doesn't make sense

Reply:
It does - all users are used to finding output files inside the main app image binary dir.
Do you find any of the files in the sysbuild folder valuable enough to pollute main image binary dir?
I find most of them as temporary/intermediate files.